NPE on Arc in combination with Myfaces extension: Producer for Object type

**Describe the bug**
When using the MyFaces Quarkus extension, the build fails with a NPE since Quarkus 1.9.0.CR1

```
[ERROR]     [error]: Build step io.quarkus.arc.deployment.ArcProcessor#generateResources threw an exception: java.lang.NullPointerException
[ERROR]     at io.quarkus.arc.processor.AsmUtilCopy.needsSignature(AsmUtilCopy.java:261)
[ERROR]     at io.quarkus.arc.processor.AsmUtilCopy.needsSignature(AsmUtilCopy.java:232)
[ERROR]     at io.quarkus.arc.processor.ClientProxyGenerator.generate(ClientProxyGenerator.java:117)
[ERROR]     at io.quarkus.arc.processor.BeanProcessor.generateResources(BeanProcessor.java:185)
[ERROR]     at io.quarkus.arc.deployment.ArcProcessor.generateResources(ArcProcessor.java:448)
[ERROR]     at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]     at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR]     at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:936)
[ERROR]     at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR]     at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR]     at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
[ERROR]     at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
[ERROR]     at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
[ERROR]     at java.lang.Thread.run(Thread.java:748)
[ERROR]     at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] -> [Help 1]
```

When debugging into BeanProcessor, ClientProxyGenerator, AsmUtilCopy, it is this producer of type Object where he is chocking:
https://github.com/apache/myfaces/blob/master/impl/src/main/java/org/apache/myfaces/cdi/JsfArtifactProducer.java#L178

He is trying to get the parent type of Object here:
https://github.com/quarkusio/quarkus/blob/daafa61cc70224f7c2987583802ab96e69c29932/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AsmUtilCopy.java#L232


**To Reproduce**
You can run this project to reproduce the exception:
https://github.com/38leinaD/jsf-reproducer

I have not tried it yet, but i assume it would happen with any producer that produces a type Object?

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Linux dev-vm.local 3.10.0-862.11.6.el7.x86_64 #1 SMP Tue Aug 14 21:49:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
 - Output of `java -version`: openjdk version "1.8.0_265"
OpenJDK Runtime Environment (build 1.8.0_265-b01)
OpenJDK 64-Bit Server VM (build 25.265-b01, mixed mode)
 - Quarkus version or git rev: 1.9.0.CR1
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Apache Maven 3.6.3

--- a/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AsmUtilCopy.java
+++ b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AsmUtilCopy.java
@@ -258,6 +258,9 @@ public class AsmUtilCopy {
      * Returns true if the given type contains parameterized types, type variables or wildcards
      */
     private static boolean needsSignature(Type type) {
+        if (type == null) {
+            return false;
+        }
         switch (type.kind()) {
             case ARRAY:
             case CLASS:

--- a/independent-projects/arc/processor/src/test/java/io/quarkus/arc/processor/AsmUtilCopyTest.java
+++ b/independent-projects/arc/processor/src/test/java/io/quarkus/arc/processor/AsmUtilCopyTest.java
@@ -1,0 +1,15 @@
+package io.quarkus.arc.processor;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import java.io.IOException;
+import org.junit.jupiter.api.Test;
+
+public class AsmUtilCopyTest {
+
+    @Test
+    public void testNeedsSignature() throws IOException {
+        assertFalse(AsmUtilCopy.needsSignature(Basics.index(Object.class).getClassByName(DotNames.OBJECT)));
+    }
+
+}

